Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -195,10 +199,79 @@ private MapPolyline parseLineString()
         skipToEndOfTag();
       }
     }
-    verifyCoordinatesParsed(hasParsedCoordinates);
+    verifyElementSeen("coordinates", hasParsedCoordinates);
     return line;
   }
 
+  @NonNull
+  private MapPolygon parsePolygon() throws IOException, XmlPullParserException, KMLParseException {
+    mParser.require(XmlPullParser.START_TAG, mNameSpace, "Polygon");
+    MapPolygon polygon = mFactory.createMapPolygon();
+    ArrayList<ArrayList<Geoposition>> rings = new ArrayList<>();
+    boolean hasOuterBoundary = false;
+    AltitudeReferenceSystemWrapper altitudeReferenceSystemWrapper =
+        new AltitudeReferenceSystemWrapper(AltitudeReferenceSystem.GEOID);
+    while (mParser.next() != XmlPullParser.END_TAG) {
+      if (mParser.getEventType() != XmlPullParser.START_TAG) {
+        continue;
+      }
+      String type = mParser.getName();
+      if (type.equals("outerBoundaryIs") || type.equals("innerBoundaryIs")) {
+        if (type.equals("outerBoundaryIs")) {
```

Comment:
Is it ok to have 2 inner boundaries? If so a comment describing why outer boundary is checked for to only occur once is helpful.

Reply:
Yes; one outer boundary is required, and 0 or more inner boundaries are allowed. I'll add a comment.